src/main.rs

```diff
@@ -652,14 +753,20 @@ fn range_names_from(
     elf: &ElfFile,
     sd: SectionData,
     text: Option<Shndx>,
-) -> Result<(RangeNames, Option<u32>), anyhow::Error> {
+) -> Result<(RangeNames, Option<u32>, bool /* uses heap */), anyhow::Error> {
     let mut range_names = vec![];
     let mut rtt = None;
+    let mut uses_heap = false;
     if let SectionData::SymbolTable32(entries) = sd {
         for entry in entries {
             if let Ok(name) = entry.get_name(elf) {
-                if name == "_SEGGER_RTT" {
-                    rtt = Some(entry.value() as u32);
+                match name {
+                    "_SEGGER_RTT" => rtt = Some(entry.value() as u32),
+                    "__rust_alloc" | "__rg_alloc" | "__rdl_alloc" | "malloc" if !uses_heap => {
```

Comment:
isn't `__rust_alloc` an implementation detail? for now it will make do though

Reply:
Yeah it is